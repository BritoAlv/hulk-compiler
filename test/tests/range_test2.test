type range(st:Number, ed:Number, offset : Number) {
    st = st;
    ed = ed;
    offset = offset;
    current = st - offset;

    next(): Boolean => (self.current := self.current + self.offset) < self.ed;
    m(): Number => self.current;
}

let X = new range(3, 6, 1) in while(X.next())
{
    print(X.m());
};