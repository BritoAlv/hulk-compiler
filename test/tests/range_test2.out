268576580
268576596
268576612

